pkg/ipam/metadata/manager.go

```diff
@@ -116,8 +140,17 @@ func (m *Manager) GetIPPoolForPod(owner string) (pool string, err error) {
 		return "", fmt.Errorf("failed to lookup pod %q: %w", namespace+"/"+name, err)
 	} else if !ok {
 		return "", &ResourceNotFound{Resource: "Pod", Namespace: namespace, Name: name}
-	} else if ipPool, hasAnnotation := pod.Annotations[annotation.IPAMPoolKey]; hasAnnotation {
-		return ipPool, nil
+	} else {
+		podAnnotations := map[string]string{
+			"ipv4Annotation": pod.Annotations[annotation.IPAMIPv4PoolKey],
+			"ipv6Annotation": pod.Annotations[annotation.IPAMIPv6PoolKey],
+			"ipAnnotation":   pod.Annotations[annotation.IPAMPoolKey],
+		}
+
+		ippool, ok := deternimePoolByAnnotations(podAnnotations, family)
```

Comment:
I'm confused. Why can't you pass in

```suggestion
		ippool, ok := deternimePoolByAnnotations(pod.Annotations, family)
```

here?

Reply:
Yes, I didn't notice that. Thank you.